doc/src/examples/algorithms/simplify_insert.cpp

```diff
@@ -18,18 +18,12 @@
 
 int main()
 {
-    typedef boost::geometry::model::d2::point_xy<double> P;
-    typedef boost::geometry::model::linestring<P> L;
-
-    L line;
-    line.push_back(P(1.1, 1.1));
-    line.push_back(P(2.5, 2.1));
-    line.push_back(P(3.1, 3.1));
-    line.push_back(P(4.9, 1.1));
-    line.push_back(P(3.1, 1.9));
+    using L = boost::geometry::model::linestring<boost::geometry::model::d2::point_xy<double>>;
 
+    L const line{{1.1, 1.1}, {2.5, 2.1}, {3.1, 3.1}, {4.9, 1.1}, {3.1, 1.9}};
     L simplified;
-    boost::geometry::simplify_inserter(line, std::back_inserter(simplified), 0.5);
+    
+    boost::geometry::detail::simplify::simplify_insert(line, std::back_inserter(simplified), 0.5);
```

Comment:
We probably should just delete all examples using `detail`, OR promote it to the general API (such as `assign_box_corners`). But not this one probably.